### ISSUE TITLE
Possible typos in makefiles

### ISSUE THREAD

**Issue by utelle** [jmcnamara/libxlsxwriter#265]
Today I tried to update my clone of libxlsxwriter to your latest master. While trying to track down why my travis CI fails for the [test_valgrind](https://travis-ci.org/utelle/libxlsxwriter/jobs/637393795?utm_medium=notification&utm_source=github_status) cases (_I get duplicate symbols on linking, and I would be grateful if could take a look why this might be so - TIA_), I found a few possible typos in your makefiles:

- line 14 in `Makefile` and `src/Makefile` defines symbol `DESTIDR` - shouldn't that be `DESTDIR`?
- line 49 in `Makefile` in the `clean` target uses symbol `USE_STANDARD_TMPFILE` to clean the minizip files - shouldn't that be symbol `USE_SYSTEM_MINIZIP`?

**Comment by jmcnamara** [comment]
Hi,

Thanks those are both typos/bugs.

I had a quick look at your fork and I don't see anything that would cause that link issue. I'll look at it in more detail and let you know.

**Comment by utelle** [comment]
> I had a quick look at your fork and I don't see anything that would cause that link issue. I'll look at it in more detail and let you know.

Great. Thanks. Most likely it is some small an subtle issue.

**Comment by jmcnamara** [comment]
I ran the test against your branch/fork and I didn't see the issue. Maybe just resubmit/rerun the test by creating a new branch or pushing a minor change.

**Comment by jmcnamara** [comment]
@sjmulder on the DESTIDR typo:

https://github.com/jmcnamara/libxlsxwriter/blob/master/Makefile#L14

**Comment by sjmulder** [comment]
@jmcnamara yes that's a typo - I can fix it or someone else can. But it doesn't break anything as-as since the line is a no-op intended for documentation (or easy tweaking).

**Comment by utelle** [comment]
> I ran the test against your branch/fork and I didn't see the issue. Maybe just resubmit/rerun the test by creating a new branch or pushing a minor change.

I restarted the failing job, but it failed again. Only the `test_valgrind` part fails, that is:
```
env:  - CFLAGS='-Werror'
script:  - make test_valgrind USE_DOUBLE_FUNCTION=1 V=1
```
I'm at a loss why the link step fails with duplicate symbols.

**Comment by utelle** [comment]
As recommended in the error message I added the linker option `-v` (see [verbose test_valgrind](https://travis-ci.org/utelle/libxlsxwriter/jobs/637633297?utm_medium=notification&utm_source=github_status)). This revealed that the object file for `utility.c` is contained 2 times in the library file, as `utility.o` and as `utility.c.o`. This seems to be a side effect of the `test_cmake` script. Not sure what's going on here.

**Comment by utelle** [comment]
As an attempt of a workaround I moved the `test_cmake` to the end of the script list. This helped a bit, but the second `test_valgrind` run still fails (see [travis CI](https://travis-ci.org/utelle/libxlsxwriter?utm_medium=notification&utm_source=github_status)).

**Comment by jmcnamara** [comment]
I was able to reproduce this. There is a memory leak running under gcc. It is a little odd but at least I can reproduce it. I'm working on a fix.

**Comment by utelle** [comment]
Thanks for looking into this.

Moving the `test_cmake` to the end of the script list is in fact only a workaround. At least in my branch the problem arises from cmake using extension `*.c.o` for object files instead of just `*.o` and adding those files to the existing library file. AFAICT this behaviour is also shown in your repository. Strange enough it does not result in errors for your CI runs.

**Comment by jmcnamara** [comment]
There are 2 issues.

1. The CMake issue, that I don't understand but will look into. Moving it to the end or adding a "make clean" after it runs should be okay. 
2. There is a valgrind warning that only occurs for some gcc/valgrind combination.

There is a fix for the valgrind warning here: https://github.com/jmcnamara/libxlsxwriter/commit/335548e77b8425a6fa3611315ab9ba8e3c4da0cb

However, I haven't bottomed out on whether this is a valid warning/error/fix. I'm still looking into it.

**Comment by jmcnamara** [comment]
I've fixed the typos and valgrind issue on master. Thanks for the report.

**Comment by utelle** [comment]
Thanks again for your excellent support.

With your modifications my library variant now passes again all CI tests.

BTW, maybe you should consider to edit the release titles of your release 0.9.2 to 0.9.4, since they reference January 2019 instead of January 2020.

**Comment by jmcnamara** [comment]
> BTW, maybe you should consider to edit the release titles of your release 0.9.2 to 0.9.4, since they reference January 2019 instead of January 2020.

Lol. Thanks.

Btw, I've started to look at adding a third party sprintf lib to fix #64.  From a code point of view my preference is for fpconv instead of emyg_dtoa but I'll test both.

**Comment by utelle** [comment]
I will take a look at `fpconv` again. As mentioned in my [post of August 10, 2017](https://github.com/jmcnamara/libxlsxwriter/issues/64#issuecomment-321398715) `fpconv` is based on the same algorithm as `emyg_dtoa`. So in principle it should work equally well.

Nevertheless, my adjusted version of `emyg_dtoa` obviously seems to work properly, otherwise my library variant wouldn't pass all the extensive CI tests of libxlsxwriter.
